src/acdl/acdl_domain.cpp

```diff
@@ -1569,6 +1569,10 @@ void acdl_domaint::normalize_val_syntactic(valuet &value)
 #ifdef DEBUG
         std::cout << "[ACDL-DOMAIN] remove_expr: " << from_expr(SSA.ns, "", m) << std::endl;
 #endif
+        // preprocess val in case it contains
+        // ID_equality. Subsumption only allows 
+        // ID_lt, ID_gt, ID_le,  ID_ge constraints
+        preprocess_val(new_val);
```

Comment:
ID_equality is handled inside preprocess_val(). This converts an expression of the type (x==N) to (x<=N) and (x>=N). The equality expressions comes from special decisions that tries to aggressively assign concrete values to variables.